### PR TITLE
added in ability to update AWS registries in other environments

### PR DESCRIPTION
**Pull request by deniseschannon**

### CODE REVIEW

**Comment by deniseschannon** [comment]
Updated from https://github.com/rancher/rancher-catalog/pull/1044

 rancher/rancher#10329

File: infra-templates/ecr/1/docker-compose.yml.tpl [hunk 1]
```diff
@@ -5,10 +5,17 @@ ecr-updater:
     AWS_REGION: ${aws_region}
     AUTO_CREATE: ${auto_create}
     LOG_LEVEL: ${log_level}
+    {{- if eq .Values.registry_in_which_environment "other" }}
+    CATTLE_URL: ${environment_api_endpoint}
+    CATTLE_ACCESS_KEY: ${environment_api_access_key}
+    CATTLE_ACCESS_SECRET: ${environment_api_secret_key}
```
Comment (superseb) [inline infra-templates/ecr/1/docker-compose.yml.tpl]: @kingsd041 Did you test with `CATTLE_ACCESS_SECRET` or with `CATTLE_SECRET_KEY`? I believe this should be `CATTLE_SECRET_KEY`

File: infra-templates/ecr/1/rancher-compose.yml [hunk 2]
```diff
@@ -32,5 +32,31 @@
       default: "INFO"
       required: true
       type: "string"
+    - variable: "registry_in_which_environment"
+      type: enum
+      label: "Registry Environment"
+      description: "Which environemnt is the AWS registry located?"
```
Comment (superseb) [inline infra-templates/ecr/1/rancher-compose.yml]: environemnt -> environment

File: infra-templates/ecr/1/rancher-compose.yml [hunk 3]
```diff
@@ -32,5 +32,31 @@
...
+      default: current
+      options:
+        - current
+        - other
+    - variable: "environment_api_endpoint"
+      label: "URL (Environment API Endpoint) of Registry to be Updated"
+      description: "URL for where the registry is located. This is the endpoint for an Environment API Key, which can be found under the Advanced Optios under API."
```
Comment (superseb) [inline infra-templates/ecr/1/rancher-compose.yml]: which can be found under API -> Keys

File: infra-templates/ecr/1/docker-compose.yml.tpl (hunk 1 above)
Comment (orangedeng) [reply infra-templates/ecr/1/docker-compose.yml.tpl]: Agree, my mistake.

File: infra-templates/ecr/1/rancher-compose.yml (hunk 3 above)
Comment (orangedeng) [reply infra-templates/ecr/1/rancher-compose.yml]: I think this is better
`URL for where the registry is located. This is the endpoint, which can be found under the Advanced Options of API page, for an Environment API Key.`

**Pull request merged**